motoman_hc20_support/urdf/hc30pl_macro.xacro

```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" ?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+  <xacro:macro name="motoman_hc30pl" params="prefix">
+    <xacro:include filename="$(find motoman_resources)/urdf/common_materials.xacro"/>
+
+    <!-- link list -->
+    <link name="${prefix}base_link">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/visual/base_link.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/base_link.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_1_s">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/visual/link_1_s.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/link_1_s.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_2_l">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/visual/link_2_l.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/link_2_l.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_3_u">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/visual/link_3_u.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/link_3_u.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_4_r">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/visual/link_4_r.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/link_4_r.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_5_b">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20dtp/visual/link_5_b.dae"/>
+        </geometry>
+        <origin rpy="0 1.57075 0"/>
```

Comment:
why do these Collada meshes have a different orientation?

It would be nice to have the meshes aligned already, so as to simplify the `xacro:macro`.

Same comment for the `collision` of `link_5_b` and both meshes for `link_6_t`.

Reply:
Because the HC30PL cannot have the B-axis in the horizontal position, so its zero position is with the wrist pointing down. I didn't want add an extra collada mesh just to turn it 90 degrees when it can be done here inthe xacro.